### DISCUSSION TITLE
Looking for advice: how to cluster 150k time series?

### DISCUSSION THREAD

**Discussion by hummuscience** [asardaes/dtwclust#75] [Q&A]
Hey!

I have used your package successfully with a small number of time-series (a few 100) and I would like to scale it now to my whole dataset (150k time series).

I have tried to run it with 15k time series, but no setting I tried seems to work (it gets stuck). Or am I missing something?

What would be the best way to approach this?

**Comment by asardaes** [comment]
Hello, you don't specify exactly which settings you're talking about. Processing so many series can indeed be a challenge, especially in R. Partitional clustering without PAM could maybe work, not sure.

**Comment by shuaiwang88** [comment]
> Hey!
> 
> I have used your package successfully with a small number of time-series (a few 100) and I would like to scale it now to my whole dataset (150k time series).
> 
> I have tried to run it with 15k time series, but no setting I tried seems to work (it gets stuck). Or am I missing something?
> 
> What would be the best way to approach this?

Did you find anything now? How many clusters do you expect for 15k time series?

**Comment by shuaiwang88** [comment]
> Hello, you don't specify exactly which settings you're talking about. Processing so many series can indeed be a challenge, especially in R. Partitional clustering without PAM could maybe work, not sure.

What's the correct arguments to begin with? I have 75k products with 3 years weekly sales data, I want to have 200 clusters or so.

**Comment by lsemployeeoftheyear** [comment]
Forgive me if I'm pirating this a bit but is anyone aware of any kind of research or studies done that might suggest how much data you actually need to train a clean cluster model? There's usually a decreasing return in investment after a certain point, if I'm not mistaken...
